flake.nix

```diff
@@ -25,6 +28,16 @@
 
     lib = extendedLib;
 
+    devShells = forAllSystems (
+      system: let
+        pkgs = nixpkgs.legacyPackages.${system};
+      in {
+        default = pkgs.mkShell {
+          packages = with pkgs; [pre-commit];
+        };
```

Comment:
We could run `pre-commit install` in the shellHook, doesn't make entering the shell much slower and ensures pre-commit is actually being used.

Reply:
Also, maybe move that to a different file ?

Reply:
> We could run `pre-commit install` in the shellHook, doesn't make entering the shell much slower and ensures pre-commit is actually being used.

Really good idea. I think it's good UX.



> Also, maybe move that to a different file ?

It can stay here, it's not a lot of code.